<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PreprocessSettingsPane</class>
 <widget class="QWidget" name="PreprocessSettingsPane">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>617</width>
    <height>653</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayoutRoot">
   <item>
    <layout class="QFormLayout" name="formLayoutPreproc">
     <item row="0" column="0">
      <widget class="QLabel" name="lblPreprocessProfile">
       <property name="text">
        <string>Input image type:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QComboBox" name="comboPreprocessProfile">
       <item>
        <property name="text">
         <string>Photo from mobile phone</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Scanner image</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Low quality</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Automatic analyzer</string>
        </property>
       </item>
      </widget>
     </item>
     <item row="1" column="0" colspan="2">
      <widget class="QLabel" name="lblPreprocessDescription">
       <property name="styleSheet">
        <string notr="true">color: #6b7280; font-size: 10pt;</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="wordWrap">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QGroupBox" name="groupAdvanced">
     <property name="title">
      <string>Advanced preprocessing parameters</string>
     </property>
     <layout class="QVBoxLayout" name="layoutAdvancedOuter">
      <item>
       <widget class="QLabel" name="lblAdvancedHint">
        <property name="styleSheet">
         <string notr="true">color: #6b7280; font-size: 10pt;</string>
        </property>
        <property name="text">
         <string>Changes affect recognition quality.
If you are unsure, use the standard profile.</string>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QScrollArea" name="scrollAdvanced">
        <property name="widgetResizable">
         <bool>true</bool>
        </property>
        <widget class="QWidget" name="scrollAdvancedContents">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>559</width>
           <height>624</height>
          </rect>
         </property>
         <layout class="QFormLayout" name="formLayoutAdvanced">
          <item row="0" column="0">
           <widget class="QLabel" name="lblShadow">
            <property name="text">
             <string>Shadow removal (shadow_removal)</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QCheckBox" name="chkShadowEnabled">
            <property name="text">
             <string>Enabled</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="lblShadowKernel">
            <property name="text">
             <string>morph_kernel</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QSpinBox" name="spinShadowMorphKernel">
            <property name="minimum">
             <number>15</number>
            </property>
            <property name="maximum">
             <number>101</number>
            </property>
            <property name="singleStep">
             <number>2</number>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="lblBg">
            <property name="text">
             <string>Background normalization (background_normalization)</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QCheckBox" name="chkBgEnabled">
            <property name="text">
             <string>Enabled</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="lblBgBlur">
            <property name="text">
             <string>blur_ksize</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QSpinBox" name="spinBgBlurKsize">
            <property name="minimum">
             <number>15</number>
            </property>
            <property name="maximum">
             <number>201</number>
            </property>
            <property name="singleStep">
             <number>2</number>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="lblBgEps">
            <property name="text">
             <string>epsilon</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QDoubleSpinBox" name="dblBgEpsilon">
            <property name="decimals">
             <number>6</number>
            </property>
            <property name="minimum">
             <double>0.000100000000000</double>
            </property>
            <property name="maximum">
             <double>1.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.001000000000000</double>
            </property>
           </widget>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="lblGauss">
            <property name="text">
             <string>Noise reduction (gaussian_blur)</string>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <widget class="QCheckBox" name="chkGaussEnabled">
            <property name="text">
             <string>Enabled</string>
            </property>
           </widget>
          </item>
          <item row="6" column="0">
           <widget class="QLabel" name="lblGaussKernel">
            <property name="text">
             <string>kernel_size</string>
            </property>
           </widget>
          </item>
          <item row="6" column="1">
           <widget class="QSpinBox" name="spinGaussKernel">
            <property name="minimum">
             <number>3</number>
            </property>
            <property name="maximum">
             <number>21</number>
            </property>
            <property name="singleStep">
             <number>2</number>
            </property>
           </widget>
          </item>
          <item row="7" column="0">
           <widget class="QLabel" name="lblGaussSigma">
            <property name="text">
             <string>sigma</string>
            </property>
           </widget>
          </item>
          <item row="7" column="1">
           <widget class="QDoubleSpinBox" name="dblGaussSigma">
            <property name="decimals">
             <number>2</number>
            </property>
            <property name="minimum">
             <double>0.100000000000000</double>
            </property>
            <property name="maximum">
             <double>5.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.100000000000000</double>
            </property>
           </widget>
          </item>
          <item row="8" column="0">
           <widget class="QLabel" name="lblClahe">
            <property name="text">
             <string>Local contrast enhancement (clahe)</string>
            </property>
           </widget>
          </item>
          <item row="8" column="1">
           <widget class="QCheckBox" name="chkClaheEnabled">
            <property name="text">
             <string>Enabled</string>
            </property>
           </widget>
          </item>
          <item row="9" column="0">
           <widget class="QLabel" name="lblClaheClip">
            <property name="text">
             <string>clip_limit</string>
            </property>
           </widget>
          </item>
          <item row="9" column="1">
           <widget class="QDoubleSpinBox" name="dblClaheClipLimit">
            <property name="decimals">
             <number>1</number>
            </property>
            <property name="minimum">
             <double>1.000000000000000</double>
            </property>
            <property name="maximum">
             <double>10.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.500000000000000</double>
            </property>
           </widget>
          </item>
          <item row="10" column="0">
           <widget class="QLabel" name="lblClaheTile">
            <property name="text">
             <string>tile_grid_size</string>
            </property>
           </widget>
          </item>
          <item row="10" column="1">
           <widget class="QSpinBox" name="spinClaheTileGrid">
            <property name="minimum">
             <number>4</number>
            </property>
            <property name="maximum">
             <number>16</number>
            </property>
            <property name="singleStep">
             <number>1</number>
            </property>
           </widget>
          </item>
          <item row="11" column="0">
           <widget class="QLabel" name="lblSharpen">
            <property name="text">
             <string>Sharpening (sharpen)</string>
            </property>
           </widget>
          </item>
          <item row="11" column="1">
           <widget class="QCheckBox" name="chkSharpenEnabled">
            <property name="text">
             <string>Enabled</string>
            </property>
           </widget>
          </item>
          <item row="12" column="0">
           <widget class="QLabel" name="lblSharpenStrength">
            <property name="text">
             <string>strength</string>
            </property>
           </widget>
          </item>
          <item row="12" column="1">
           <widget class="QDoubleSpinBox" name="dblSharpenStrength">
            <property name="decimals">
             <number>2</number>
            </property>
            <property name="minimum">
             <double>0.000000000000000</double>
            </property>
            <property name="maximum">
             <double>2.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.100000000000000</double>
            </property>
           </widget>
          </item>
          <item row="13" column="0">
           <widget class="QLabel" name="lblSharpenGK">
            <property name="text">
             <string>gaussian_ksize</string>
            </property>
           </widget>
          </item>
          <item row="13" column="1">
           <widget class="QSpinBox" name="spinSharpenGaussianK">
            <property name="minimum">
             <number>3</number>
            </property>
            <property name="maximum">
             <number>21</number>
            </property>
            <property name="singleStep">
             <number>2</number>
            </property>
           </widget>
          </item>
          <item row="14" column="0">
           <widget class="QLabel" name="lblSharpenGS">
            <property name="text">
             <string>gaussian_sigma</string>
            </property>
           </widget>
          </item>
          <item row="14" column="1">
           <widget class="QDoubleSpinBox" name="dblSharpenGaussianSigma">
            <property name="decimals">
             <number>2</number>
            </property>
            <property name="minimum">
             <double>0.100000000000000</double>
            </property>
            <property name="maximum">
             <double>5.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.100000000000000</double>
            </property>
           </widget>
          </item>
          <item row="15" column="0">
           <widget class="QLabel" name="lblAdaptive">
            <property name="text">
             <string>Adaptive thresholding (adaptive_threshold)</string>
            </property>
           </widget>
          </item>
          <item row="15" column="1">
           <widget class="QCheckBox" name="chkAdaptiveEnabled">
            <property name="text">
             <string>Enabled</string>
            </property>
           </widget>
          </item>
          <item row="16" column="0">
           <widget class="QLabel" name="lblAdaptiveBlock">
            <property name="text">
             <string>block_size</string>
            </property>
           </widget>
          </item>
          <item row="16" column="1">
           <widget class="QSpinBox" name="spinAdaptiveBlockSize">
            <property name="minimum">
             <number>11</number>
            </property>
            <property name="maximum">
             <number>101</number>
            </property>
            <property name="singleStep">
             <number>2</number>
            </property>
           </widget>
          </item>
          <item row="17" column="0">
           <widget class="QLabel" name="lblAdaptiveC">
            <property name="text">
             <string>C</string>
            </property>
           </widget>
          </item>
          <item row="17" column="1">
           <widget class="QSpinBox" name="spinAdaptiveC">
            <property name="minimum">
             <number>-20</number>
            </property>
            <property name="maximum">
             <number>20</number>
            </property>
            <property name="singleStep">
             <number>1</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacerBottom">
     <property name="orientation">
      <enum>Qt::Orientation::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>20</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
